build-script.sh

```diff
@@ -50,7 +50,7 @@ for dir in $DIRS; do
     fi
   else
     # install v4 dependencies
-    go get ./...
+    go get -u go.opentelemetry.io/otel
```

Comment:
My PR to opentelemetry is included in v0.10.0, however doing a `go get ./...` installs what's currently listed in the go.mod file, which is v0.5.0.

Reply:
Why don't we update the `go.mod` file to `0.10.0`?

Reply:
With the change in the `go.mod` file, do we still need the `go get -u go.opentelemetry.io/otel` line?

Reply:
Yeah, I was thinking about that.

With `go get -u go.opentelemetry.io/otel` we will always test against the latest version of otel.

With `go get ./...` we will test against whatever is written in the go mod file (I think, this might depend on go version, so I'm not 100% certain).

Personally, I'd rather test against the latest version of the otel package so we can catch changes as quickly as we can. However, then we could end up being out of sync with our go.mod file.

So basically, I'm not sure what the right answer is. What do you think?

Reply:
I'll try to imagine the worst possible scenario:
1. Our `go.mod` file references an older version `0.x`.
2. Our tests pull the newest version `0.y`.
3. We modify our code to work with `0.y` but don't update the `go.mod` file. Unfortunately, those are breaking changes, so the agent won't work anymore with `0.x`.
4. We release.
5. It breaks for customers.

I'm not 100% familiar with all the `go.mod` delicacies, but I'd opt for running the tests with the same version we have in `go.mod`.

Reply:
Very good point. I'll make the update!